src/main/java/com/minecolonies/util/StructureWrapper.java

```diff
@@ -234,31 +197,28 @@ else if (localBlock.getMaterial().isSolid())
                     }
 
                     //setTileEntity checks for null and ignores it.
-                    world.setTileEntity(worldPos, schematicWorld.getTileEntity(localPos));
+                    world.setTileEntity(worldPos, structure.getTileEntity(localPos));
                 }
             }
         }
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'coords' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)